python/hail/dataset.py

```diff
@@ -364,6 +364,54 @@ def annotate_alleles_expr(self, expr, propagate_gq=False):
         jvds = self._jvdf.annotateAllelesExpr(expr, propagate_gq)
         return VariantDataset(self.hc, jvds)
 
+
+    @handle_py4j
+    def annotate_alleles_vds(self, other, code, match_star = True):
+        """Annotate variants / alleles from another VDS
+
+        **Examples**
+
+        Import a second VDS, ``vd2``, with annotations to merge into ``vds``:
+
+        >>> vds2 = hc.read("data/example2.vds")
+
+        Copy the inbreeding coefficient annotation (``Double``) for each site from ``vds2`` ``va.info.InbreedingCoeff`` to ``vds`` ``va.info.other_InbreedingCoeff``:
+
+        >>> vds1 = vds.annotate_alleles_vds(vds2, 'va.info.other_InbreedingCoeff = vds.find(x => isDefined(x)).info.InbreedingCoeff')
+
+        Copy allele frequencies (``Array[Double]`` containing the allele frequency for each alternative allele) for each allele from
+        ``vds2`` ``va.info.AF`` to ``vds`` ``va.info.other_AF``:
+
+        >>> vds1 = vds.annotate_alleles_vds(vds2, 'va.info.other_AF = range(v.nAltAlleles).map(i => vds[i].info.AF[aIndices[i]])')
+
+        **Notes**
+
+        The ``code`` argument expects an annotation expression whose scope includes ``va``, the variant annotations in the current VDS,
+        and ``vds``, the variant annotations in other (one per allele).
+        This method matches alleles based on their position, allowing to annotate a VDS that is split or non-split with
+        another VDS that is split or non-split, as long as the allele representation is consistently aligned (e.g. GATK HaplotypeCaller alleles are all left-aligned).
+        The alleles at each position are matched between the VDSs.
+
+        ``code`` is in a variant context and the following symbols are in scope:
+
+           - ``v`` (*Variant*): :ref:`variant`
+           - ``va``: variant annotations
+           - ``vds`` (*Array*): variant annotations from ``other``. There is one entry for each allele in ``v``, entries contain ``NA`` if the allele was not found in ``other``.
+           - ``aIndices`` (*Array[Int]*): The mapping from the alt allele index in the VDS and those in ``other``. If the other VDS was split, the entries are ``0`` for alleles that match. There is one entry for each allele in ``v``, entries contain ``NA`` if the allele was not found in ``other``.
+           - ``global``: global annotations
+
+        :param VariantDataset other: Variant dataset to annotate with.
```

Comment:
Use :py:class: reference here. It needs to be on a separate line with `:type other: ...`. See elsewhere in the docs